chia/full_node/full_node.py

```diff
@@ -2481,6 +2487,34 @@ async def add_compact_vdf(self, request: full_node_protocol.RespondCompactVDF, p
         if self._server is not None:
             await self.server.send_to_all([msg], NodeType.FULL_NODE, peer.peer_node_id)
 
+    def check_bad_peak_cache(self, wp: WeightProof) -> bool:
+        for block in wp.recent_chain_data:
+            if block.header_hash in self.bad_peak_cache.keys():
+                return False
+        return True
+
+    def add_to_bad_peak_cache(self, peak_header_hash: bytes32, peak_height: uint32) -> None:
+        curr_height = self.blockchain.get_peak_height()
+
+        if curr_height is None:
+            self.log.debug(f"add bad peak {peak_header_hash} to cache")
+            self.bad_peak_cache[peak_header_hash] = peak_height
+            return
+
+        if peak_height < curr_height - 2 * self.constants.SUB_EPOCH_BLOCKS:
```

Comment:
i think this is more readable, a name like "last_two_epoch_height" would be misleading since this is the max possible size of the two epochs not necessarily where they started, if you have a good name i dont mind changing

Reply:
I don't need to make a big deal here, but I'll make an attempt with `minimum_cache_height`?

Reply:
ok thats a good one ill take it thanks